packages/firestore/src/core/sync_engine.ts

```diff
@@ -238,7 +240,7 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
         targetId,
         status === 'current'
       );
-      if (this.isPrimary) {
+      if (this.isPrimaryClient) {
```

Comment:
It seems like this factoring leaves a lot of multi-tab knowledge in the base SyncEngine. Looking at how this logic breaks down I can't say I have any great ideas for how to improve that though.